src/core/Atoms.cpp

```diff
@@ -138,54 +137,81 @@ void Atoms::setForces(void*p,int i) {
 }
 
 void Atoms::share() {
-  std::set<AtomNumber> unique;
 // At first step I scatter all the atoms so as to store their mass and charge
 // Notice that this works with the assumption that charges and masses are
 // not changing during the simulation!
   if(!massAndChargeOK && shareMassAndChargeOnlyAtFirstStep) {
     shareAll();
     return;
   }
-  for(unsigned i=0; i<actions.size(); i++) if(actions[i]->isActive()) {
-      if(dd && shuffledAtoms>0) {
-        unique.insert(actions[i]->getUnique().begin(),actions[i]->getUnique().end());
+
```

Comment:
shuffledAtoms means that the local index of the atoms does not correspond to the global one